core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala

```diff
@@ -218,16 +220,26 @@ class KafkaConfigTest {
     props.put(KafkaConfig.ZkConnectProp, "localhost:2181")
 
     // listeners with duplicate port
-    props.put(KafkaConfig.ListenersProp, "PLAINTEXT://localhost:9091,TRACE://localhost:9091")
-    assertFalse(isValidKafkaConfig(props))
+    props.put(KafkaConfig.ListenersProp, "PLAINTEXT://localhost:9091,SSL://localhost:9091")
+    var caught = intercept[IllegalArgumentException] { KafkaConfig.fromProps(props) }
+    assertThat(caught.getMessage(), containsString("Each listener must have a different port"))
```

Comment:
Can we use something like:
```
assertTrue(caught.getMessage.contains("Each listener must have a different port"))
```
That way we don't need to import hamcrest and only use junit.

This can be applied to all `assertThat()` added in this PR

Reply:
Nice. I'll change that.